Pcap++/src/NetworkUtils.cpp

```diff
@@ -337,15 +337,17 @@ IPv4Address NetworkUtils::getIPv4Address(const std::string& hostname, PcapLiveDe
 		dnsTimeout = NetworkUtils::DefaultTimeout;
 
 	// validate DNS server IP. If it wasn't provided - set the system-configured DNS server
-	if (dnsServerIP == IPv4Address::Zero && device->getDnsServers().size() > 0)
-	{
-		dnsServerIP = device->getDnsServers().at(0);
-	}
-
-	if (!dnsServerIP.isValid())
+	if (dnsServerIP == IPv4Address::Zero)
 	{
-		PCPP_LOG_ERROR("DNS server IP isn't valid");
-		return result;
+		if (device->getDnsServers().size() > 0)
+		{
+			dnsServerIP = device->getDnsServers().at(0);
+		}
+		else
+		{
+			PCPP_LOG_ERROR("DNS server IP wasn't provided and couldn't find system-configured DNS server");
+			return result;
+		}
 	}
```

Comment:
Why do we need to change this logic? We can do:
```cpp
if (dnsServerIP == IPv4Address::Zero && device->getDnsServers().size() > 0)
{
	dnsServerIP = device->getDnsServers().at(0);
}
```

Reply:
what if `dnsServerIP` is ZERO and `device->getDnsServers().size()` is 0?

Reply:
That's a good point! We can do:
```cpp
if (dnsServerIP == IPv4Address::Zero && device->getDnsServers().size() > 0)
{
	dnsServerIP = device->getDnsServers().at(0);
}

if (dnsServerIP == IPv4Address::Zero)
{
	PCPP_LOG_ERROR("DNS server IP wasn't provided and couldn't find system-configured DNS server");
	return result;
}
```

Reply:
sure.